appsec-kit-v24/src/main/java/com/vaadin/appsec/v24/views/DependenciesTab.java

```diff
@@ -0,0 +1,208 @@
+
+        getListDataProvider().setFilter(dependencyDTO -> {
+            if (ecosystemFilter != null
+                    && !ecosystemFilter.equals(dependencyDTO.getEcosystem())) {
+                return false;
+            }
+            if (groupFilter != null
+                    && !groupFilter.equals(dependencyDTO.getGroup())) {
+                return false;
+            }
+            if (includeNpmDevDeps && isDevelopmentFilter != null
+                    && !isDevelopmentFilter == dependencyDTO
+                            .isDevDependency()) {
+                return false;
+            }
+            if (severityFilter != null && !severityFilter
+                    .equals(dependencyDTO.getSeverityLevel())) {
+                return false;
+            }
+            return riskScoreFilter == null
+                    || riskScoreFilter <= dependencyDTO.getRiskScore();
+        });
+    }
+
+    @Override
+    public void refresh() {
+        dataView = grid.setItems(AppSecService.getInstance().getDependencies());
+        dataView.addFilter(dependency -> {
+            String searchTerm = searchField.getValue().trim();
+            if (searchTerm.isEmpty()) {
+                return true;
+            }
+            return dependency.getName().contains(searchTerm);
+        });
+
+        List<String> sortedGroups = getListDataProvider().getItems().stream()
+                .map(Dependency::getGroup).filter(Objects::nonNull).distinct()
+                .sorted().toList();
+        group.setItems(sortedGroups);
+        applyFilters();
+    }
+
+    private Double getRiskScoreFromFilter(String riskScoreFilter) {
+        String[] parts = riskScoreFilter.split("=");
+        return Double.valueOf(parts[1]);
+    }
+
+    private void configureSearchField() {
+        dataView = grid.setItems(AppSecService.getInstance().getDependencies());
+
+        searchField.setPlaceholder("Search");
+        searchField.setPrefixComponent(new Icon(VaadinIcon.SEARCH));
+        searchField.setValueChangeMode(ValueChangeMode.EAGER);
+        searchField.addValueChangeListener(e -> dataView.refreshAll());
+    }
+
+    private void buildFilters() {
+        searchField = new TextField("Dependency name");
+
+        ecosystem = new ComboBox<>("Ecosystem");
+        ecosystem.setItems(Ecosystem.MAVEN, Ecosystem.NPM);
+
+        group = new ComboBox<>("Dependency group");
+
+        if (includeNpmDevDeps) {
+            isDevelopment = new ComboBox<>("Is development?");
+            isDevelopment.setItems(Boolean.TRUE, Boolean.FALSE);
+        }
+
+        severity = new ComboBox<>("Severity");
+        severity.setItems(SeverityLevel.NONE, SeverityLevel.LOW,
+                SeverityLevel.MEDIUM, SeverityLevel.HIGH,
+                SeverityLevel.CRITICAL);
+
+        riskScore = new ComboBox<>("CVSS score");
+        riskScore.setItems(">= 0", ">=1", ">=2", ">=3", ">=4", ">=5", ">=6",
+                ">=7", ">=8", ">=9", ">=10");
```

Comment:
```suggestion
        riskScore.setItems(">=0", ">=1", ">=2", ">=3", ">=4", ">=5", ">=6",
                ">=7", ">=8", ">=9", ">=10");
```

Reply:
Updated. Also in the `VulnerabilitiesTab`.